File: S3/DS/binarysearch.c
// Experiment 9 - Binary Search 

#include <stdio.h>
int first, last, key;
int arr[10];
void binarysearch(int arr[], int first, int last, int key);

void main() {
	int n, i, j, t;
	printf("\nEnter the number of elements: ");
	scanf("%d", &n);
	for (i = 0; i < n; i++) {
		printf("\nEnter element %d: ", i+1);
		scanf("%d", &arr[i]);
	}
	for (i = 0; i < n-1; i++) {
		for (j = 0; j < n-1-i; j++) {
			if (arr[j] > arr[j+1]) {
				t = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = t;
			}
		}
	}
	printf("\nEnter key: ");
	scanf("%d", &key);
	binarysearch(arr, 0, n-1, key);
}

void binarysearch(int arr[], int first, int last,int key) {
	int mid = (first + last) / 2;
	if (last >= first) {
		if (key == arr[mid]) {
			printf("\nElement found at %d", mid);
			return;
		} else if (key > arr[mid]) {
			first = mid + 1;
			binarysearch(arr, first, last, key);
		} else {
			last = mid - 1;
			binarysearch(arr, first, last, key);
		}
		printf("\n");
	} else {
		printf("Element not found \n");
	}
}

File: S3/OOPS/binarysearch.java
import java.util.*;

class binarysearch {
	int binarySearch(int arr[], int l, int r, int x) {
		if (r >= 1) {
			int mid = l + (r - l) / 2;
			
			if (arr[mid] == x) {
				return mid;
			}
			
			if (arr[mid] > x) {
				return binarySearch(arr, l, mid - 1, x);
			} else {
				return binarySearch(arr, mid + 1, r, x);
			}
			}
			return -1;
			}
			
public static void main (String[] args) {
	Scanner sc = new Scanner(System.in);
	binarysearch ob = new binarysearch();
	
	System.out.println("Enter number of elements: ");
	int n = sc.nextInt();
	
	int arr[] = new int[n];
	System.out.println("Enter elements: ");
	for (int i = 0; i < n; i++) {
		arr[i] = sc.nextInt();
	}
	
	System.out.println("Enter element to be searched: ");
	int x = sc.nextInt();
	
        int result = ob.binarySearch(arr, 0, n - 1, x);
        if (result == -1)
            System.out.println("Element not present");
        else
            System.out.println("Element found at index "
                               + result);
    }
}